keps/sig-storage/4958-csi-sidecars-all-in-one/README.md

```diff
@@ -0,0 +1,1161 @@
+|  Integration with CSI Drivers    |   <li> Less config in the controller/node yaml manifest <li>Less confusion for CSI Driver authors on which CSI Sidecar versions to use @msau42  |   <li> Complex configuration for the single CSI Sidecar component <li> Difficulty expressing per CSI Sidecar configuration  e.g. kube-api-qps, kube-qpi-burst <ul><li> global flag, override through a CSI sidecar flag  e.g. kube-api-qps -> attacher-kube-api-qps | 
+
+
+
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+
+#### Glossary
+- Individual repository:  An existing repository in the kubernetes-csi/ org in Github e.g. the external-attacher repository.
+- AIO monorepo(monorepo): The monolithic repository where most of the code of the CSI Sidecars will be migrated.
+- Monorepo component:  The source code of an individual repository that is currently being migrated or already migrated to the monorepo. 
+- AIO Sidecar Image: The All-In-One sidecar image utilizes a monorepo
+- repository root path: The portion of a module path that corresponds to a version control repository’s root directory.
+
+
+#### AIO Monorepo 
+
+##### Release Management
+
+Use Semantic Versioning in monorepo.
+
+[Alternatives](#releasemanagement)
+
+
+##### RBAC policy
+
+We designed the AIO monorepo's RBAC policy to mirror that of individual repos, where each controller maintains its own policy. Driver maintainers should apply proper RBAC when enabling specific controllers in AIO
```

Comment:
This also matches what kcm has, where each controller has very narrow RBAC rules that explicitly allow it to performed desired actions, and nothing more.